collector/config.go

```diff
@@ -62,6 +62,39 @@ type Prometheus struct {
 	MetricsConfig []string `json:"metrics_config"`
 }
 
+type Jolokia struct {
+	//the endpoint to hit to scrape metrics. Eg https://${HOST}/jolokia
+	Endpoint EndpointConfig `json:"endpoint"`
+
+	//holds names of different metrics that can be collected
+	MetricsConfig []JolokiaMetricConfig `json:"metrics_config"`
+
+	//the frequency at which metrics should be collected
+	PollingFrequency string `json:"polling_frequency"`
+}
+
+type JolokiaMetricConfig struct {
+	//the name of the metric Eg "HeapMemoryUsage" or "MyMetricFoo"
+	Name string `json:"name"`
+
+	//the name of the mbean. Eg java.lang:type=Memory
+	MBean string `json:"mbean"`
+	// the attribute from the mbean Eg HeapMemoryUsage
+	Attribute string `json:"attribute"`
+	//the path contianing the mbean to be used Eg used
```

Comment:
Yeah. I can review it. It will have to wait until next week though.

On Fri, Jul 22, 2016 at 6:30 AM, Matthew Wringe <notifications@github.com>
wrote:

> In collector/config.go
> <https://github.com/google/cadvisor/pull/1392#discussion_r71878201>:
>
> > +	//holds names of different metrics that can be collected
> > +	MetricsConfig []JolokiaMetricConfig `json:"metrics_config"`
> > +
> > +	//the frequency at which metrics should be collected
> > +	PollingFrequency string `json:"polling_frequency"`
> > +}
> > +
> > +type JolokiaMetricConfig struct {
> > +	//the name of the metric Eg "HeapMemoryUsage" or "MyMetricFoo"
> > +	Name string `json:"name"`
> > +
> > +	//the name of the mbean. Eg java.lang:type=Memory
> > +	MBean string `json:"mbean"`
> > +	// the attribute from the mbean Eg HeapMemoryUsage
> > +	Attribute string `json:"attribute"`
> > +	//the path contianing the mbean to be used Eg used
>
> Ah, I can add quotes around the 'used' to make it seem more clear that the
> name of the attribute is 'used' (the used amount of HeapMemoryUsage)
>
> —
> You are receiving this because you were assigned.
> Reply to this email directly, view it on GitHub
> <https://github.com/google/cadvisor/pull/1392/files/e3625ca1cdfe4556b97e13e4fda1ab5db1ca6ac1#r71878201>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AGvIKPkjiO9eLXihAt7pabNmoI0Mkt3rks5qYMXdgaJpZM4JSNiv>
> .
>